src/IECore/Primitive.cpp

```diff
@@ -107,6 +107,54 @@ void Primitive::save( IECore::Object::SaveContext *context ) const
 	}
 }
 
+namespace
+{
+
+void convertUVs( PrimitiveVariableMap &variables )
+{
+	std::set<Data *> visited;
+
+	auto tIt = variables.find( "t" );
+	if( tIt != variables.end() )
+	{
+		if( FloatVectorData *values = runTimeCast<FloatVectorData>( tIt->second.data.get() ) )
+		{
+			for( auto &value : values->writable() )
+			{
+				value = 1.0f - value;
+			}
+		}
+
+		visited.insert( tIt->second.data.get() );
+	}
+
+	// by convention, PrimitiveVariables ending with "_t"
+	// represent the second component of a UV set.
+	for( auto it = variables.begin(), eIt = variables.end(); it != eIt; ++it )
+	{
+		if( visited.find( it->second.data.get() ) != visited.end() )
+		{
+			continue;
+		}
+
+		size_t suffixOffset = it->first.rfind( "_t" );
+		if ( ( suffixOffset != std::string::npos) && ( suffixOffset == it->first.length() - 2 ) )
+		{
+			if( FloatVectorData *values = runTimeCast<FloatVectorData>( it->second.data.get() ) )
```

Comment:
Since this transformation is identical to the one above for the "t" case, and it's about to get more complex when we start converting to indexed V2fs, how about we just use one loop over the variables, looking for `it->first == "t" || boost::ends_with( it->first, "_t" )`, so we're not duplicating logic? Either way, I really recommend `boost::algorithm::ends_with` instead of `rfind( "_t" ) == length() - 2`, it's way more readable.

Reply:
> How about we just use one loop over the variables

Sure

> Either way, I really recommend boost::algorithm::ends_with instead of rfind( "_t" )

I cribbed that from [ToMayaMeshConverter](https://github.com/ImageEngine/cortex/blob/6fa38cea55d5e8bde8dec85e8ae707a757712bec/src/IECoreMaya/ToMayaMeshConverter.cpp#L421), which I had assumed was done for a performance reason. Happy to change it though.

Reply:
Addressed in b4ea35d